contracts/src/FunctionGateway.sol

```diff
@@ -2,246 +2,270 @@
         bytes memory _input,
-        bytes4 _callbackSelector,
         bytes memory _context,
-        uint256 _gasLimit,
-        address _refundAccount
-    ) public payable returns (bytes32) {
+        bytes4 _callbackSelector,
+        uint32 _callbackGasLimit
+    ) external payable returns (bytes32) {
+        // Compute the callback hash uniquely associated with this request.
         bytes32 inputHash = sha256(_input);
-        bytes32 contextHash = keccak256(_context);
-        FunctionRequest memory r = FunctionRequest({
-            functionId: _functionId,
-            inputHash: inputHash,
-            outputHash: bytes32(0),
-            contextHash: contextHash,
-            callbackAddress: msg.sender,
-            callbackSelector: _callbackSelector,
-            proofFulfilled: false,
-            callbackFulfilled: false
-        });
-
-        uint256 feeAmount = _handlePayment(_gasLimit, _refundAccount, msg.sender, msg.value);
-
-        bytes32 requestId = keccak256(abi.encode(nonce, r));
-        requests[requestId] = r;
-
-        emit ProofRequested(nonce, _functionId, requestId, _input, _context, _gasLimit, feeAmount);
-        nonce++;
-        return requestId;
-    }
+        bytes32 contextHash = sha256(_context);
+        address callbackAddress = msg.sender;
+        bytes32 requestHash = _requestHash(
+            nonce,
+            _functionId,
+            inputHash,
+            contextHash,
+            callbackAddress,
+            _callbackSelector,
+            _callbackGasLimit
+        );
 
-    /// @dev The entrypoint for fulfilling proofs which are not in batches.
-    /// @param _requestId The id of the request to be fulfilled.
-    /// @param _outputHash The output hash of the proof.
-    /// @param _proof The proof.
-    function fulfill(bytes32 _requestId, bytes32 _outputHash, bytes memory _proof) external {
-        // Do some sanity checks.
-        FunctionRequest storage r = requests[_requestId];
-        if (r.callbackAddress == address(0)) {
-            revert RequestNotFound(_requestId);
-        } else if (r.proofFulfilled) {
-            revert ProofAlreadyFulfilled(_requestId);
-        }
+        // Store the callback hash.
+        requests[nonce] = requestHash;
+        emit Request(
+            nonce,
+            _functionId,
+            _input,
+            _context,
+            _callbackSelector,
+            _callbackGasLimit
+        );
 
-        // Update the request.
-        r.proofFulfilled = true;
-        r.outputHash = _outputHash;
+        // Increment the nonce.
+        nonce++;
 
-        // Verify the proof.
-        address verifier = verifiers[r.functionId];
-        if (!IFunctionVerifier(verifier).verify(r.inputHash, _outputHash, _proof)) {
-            revert InvalidProof(address(verifier), r.inputHash, _outputHash, _proof);
-        }
+        // Send the fee to the vault.
+        IFeeVault(feeVault).depositNative{value: msg.value}(callbackAddress);
```

Comment:
While we're at it, can we change the `FeeVault` so that it doesn't try to refund and fail if it isn't able to? It always borks out forge simulations and is super annoying